### DISCUSSION TITLE
I have a sequence of graphs which represent states of the same entity over time - should I use TGN or RENet?

### DISCUSSION THREAD

**Discussion by krzysztoffiok** [pyg-team/pytorch_geometric#2988] [Q&A]
Hi,

I'm working with fMRI data. The data can be represented as a graph with feature values changing over time of the fMRI recording session (let's assume the graph could change it's structure as well). It is possible to obtain a sequence of graphs representing the same subject and his brain states that change during the recording session. Now I'm looking for an architecture that could be used with such data/graph sequences. Do you think TGN or RENet are suitable or should I look elsewhere?

I will appreciate any thoughts on the matter!

Thanks,
Chris

**Comment by rusty1s** [comment]
I think you can start a bit more simple. Given that your data is temporal, you can create a dataset in which each data objects represents the graph at a given timestamp. Iterating over the dataset using a dataloader with `shuffle=False` gives you a batch graphs at consecutive timestamps. You can input this graph into a GNN, split the resulting node embeddings according to time, and apply a RNN/Transformer on the resulting temporal sequence. WDYT?

**Reply by krzysztoffiok** [reply]
Thanks for answering so quickly.

I guess this is simpler and I like the concept. I will have to think about how to do this and what I get at which step, but I will give it a try.

Do you think I should go with batch_size=1 and extract those node embeddings for each graph separately?

**Reply by rusty1s** [reply]
I think you want to do that in an end-to-end fashion where you first learn node embeddings on each graph individually via a GNN, and then merge them together afterwards via a sequence encoder model.

**Reply by krzysztoffiok** [reply]
OK thanks again I will work on that and let you know how it goes.

**Reply by krzysztoffiok** [reply]
Hi @rusty1s, isn't this https://github.com/benedekrozemberczki/pytorch_geometric_temporal package the one I should be using for temporal graphs like in my case? It seems they have many nice solutions implemented already!

**Reply by rusty1s** [reply]
Yes, if it fits your use-case feel free to try it out :)
